File: content-solutions/stack/vault-brackets.js
function solution(code) {
  const pair = { ")": "(", "]": "[", "}": "{" };
  const stack = [];
  for (const c of code) {
    if ("([{".includes(c)) stack.push(c);
    else if (stack.length === 0 || stack.pop() !== pair[c]) return false;
  }
  return stack.length === 0;
}

File: content-solutions/stack/vault-brackets.java
import java.util.*;

class Solution {
    public boolean solution(String code) {
        Deque<Character> stack = new ArrayDeque<>();
        for (char c : code.toCharArray()) {
            if (c == '(' || c == '[' || c == '{') {
                stack.push(c);
                continue;
            }
            char open = c == ')' ? '(' : c == ']' ? '[' : '{';
            if (stack.isEmpty() || stack.pop() != open) return false;
        }
        return stack.isEmpty();
    }
}
